File: GarbageV3/app/src/main/java/dk/itu/garbagev3/AddActivity.java
package dk.itu.garbagev3;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class AddActivity extends AppCompatActivity {

    //Model: Database of items
    private static ItemsDB itemsDB;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.add_items);

        itemsDB= ItemsDB.get();

        //Text Fields
        TextView newWhat= findViewById(R.id.what_text);
        TextView newWhere= findViewById(R.id.where_text);

        Button addItem= findViewById(R.id.add_button);
        // adding a new thing
        addItem.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String whatS= newWhat.getText().toString().trim();
                String whereS= newWhere.getText().toString().trim();
                if ((!whatS.isEmpty()) && (!whereS.isEmpty())) {
                    itemsDB.addItem(whatS, whereS);
                    newWhat.setText("");
                    newWhere.setText("");
                } else Toast.makeText(AddActivity.this, R.string.add_toast, Toast.LENGTH_LONG).show();
            }
        });

    }

}

File: KotlinGarbageV4/app/src/main/java/dk/itu/kotlingarbagev4/AddActivity.kt
package dk.itu.kotlingarbagev4

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

import androidx.appcompat.app.AppCompatActivity

class AddActivity : AppCompatActivity() {
    private lateinit var itemsDB: ItemsDB

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.add_items)
        itemsDB = ItemsDB.get()

        val newWhat: TextView = findViewById(R.id.what_text)
        val newWhere: TextView = findViewById(R.id.where_text)

        val addItem: Button = findViewById(R.id.add_button)

        addItem.setOnClickListener {
            val whatS = newWhat.text.toString().trim()
            val whereS = newWhere.text.toString().trim()
            if ((whatS.isNotEmpty()) && (whereS.isNotEmpty())) {
                itemsDB.addItem(whatS, whereS)
                newWhat.text = ""
                newWhere.text = ""
            } else {
                Toast.makeText(this, R.string.add_toast, Toast.LENGTH_LONG).show()
            }
        }
    }
}